app: Enable DTS partitions (#284)
nrfconnect/ncs-serial-modem: doc/app/sm_build_and_run.rst

When the `LwM2M carrier`_ library is in use, by default the application will auto-connect to the network on startup.
This behavior can be changed by disabling the :ref:`CONFIG_SM_AUTO_CONNECT <CONFIG_SM_AUTO_CONNECT>` option.

.. _sm_build_disable_b0:

Building without the immutable B0 bootloader
********************************************

By default, the |SM| application uses an immutable bootloader B0 (NSIB) and an upgradeable MCUboot bootloader.

Version 1.x of the |SM| application uses only an upgradeable MCUboot bootloader.

> Building without the B0 bootloader is offered as backwards compatibility option if you want to maintain the same partition layout as in version 1.x.

Reviewer: ```suggestion
Version v1.x.x of the |SM| application uses only an upgradeable MCUboot bootloader.
Building without the B0 bootloader is offered as backwards compatibility option if you want to maintain the same partition layout as in |SM| v1.x.x.
```

When the `LwM2M carrier`_ library is in use, by default the application will auto-connect to the network on startup.
This behavior can be changed by disabling the :ref:`CONFIG_SM_AUTO_CONNECT <CONFIG_SM_AUTO_CONNECT>` option.

.. _sm_build_disable_b0:

Building without the immutable B0 bootloader
********************************************

> By default, the |SM| application uses an immutable bootloader B0 (NSIB) and upgradeable MCUboot bootloader.

Reviewer 2: ```suggestion
By default, the |SM| application uses an immutable bootloader B0 (NSIB) and an upgradeable MCUboot bootloader.
```

Version 1.x of the |SM| application used only and upgradeable MCUboot bootloader.

> Building without B0 bootloader is offered as backwards compatibility option for users who want to maintain the same partition layout as in version 1.x.

Reviewer 2: ```suggestion
Version 1.x of the |SM| application uses only an upgradeable MCUboot bootloader.
Building without the B0 bootloader is offered as backwards compatibility option if you want to maintain the same partition layout as in version 1.x.
```
Considering the user is reading the document.

To build without B0 and with a single, non-updatable MCUboot slot, include
:file:`overlay-disable-b0.overlay` in both the application and MCUboot images and disable the

> secure boot appcore in the sysbuild configuration:

Reviewer 2: ```suggestion
To build without B0 and with a single, non-updatable MCUboot slot, include :file:`overlay-disable-b0.overlay` in both the application and MCUboot images and disable the secure boot appcore in the sysbuild configuration:
```

.. code-block:: console

   west build -p -b nrf9151dk/nrf9151/ns -- -DEXTRA_DTC_OVERLAY_FILE=overlay-disable-b0.overlay -Dmcuboot_EXTRA_DTC_OVERLAY_FILE=<path-to-app>/overlay-disable-b0.overlay -DSB_CONFIG_SECURE_BOOT_APPCORE=n

> Where ``<path-to-app>`` is the **absolute** path to the :file:`app` directory.

Reviewer 2: ```suggestion
Where ``<path-to-app>`` is the absolute path to the :file:`app` directory.
```
